quetz/tasks/workers.py

```diff
@@ -143,21 +151,38 @@ async def wait(self):
 
 
 class RQManager(AbstractWorker):
-    def __init__(self, host, port, db, api_key: str, browser_session: dict):
+    def __init__(
+        self,
+        host,
+        port,
+        db,
+        api_key: str,
+        browser_session: dict,
+        config: Config,
+        no_testing=True,
+    ):
         self.host = host
         self.port = port
         self.db = db
         self.api_key = api_key
         self.browser_session = browser_session
+        self.config = config
         self.conn = redis.StrictRedis(host=self.host, port=self.port, db=self.db)
-        self.queue = Queue(connection=self.conn)
+        self.queue = Queue(connection=self.conn, is_async=no_testing)
 
     def execute(self, func, *args, **kwargs):
         self.job = self.queue.enqueue(
-            wrapper, func, self.api_key, self.browser_session, *args, **kwargs
+            job_wrapper,
+            func,
+            self.api_key,
+            self.browser_session,
+            self.config,
+            *args,
+            **kwargs,
         )
 
-    def wait(self):
+    async def wait(self):
         while not self.job.is_finished:
             time.sleep(1)
-        return self.job.result
+        if self.job.result:
+            return await self.job.result
```

Comment:
you can't await results, it's not a asynchronous function, but the result of the execution (`None` in this case).

Reply:
This is done to make it compatible with the `tests`. So that the statement `await any_worker.wait()` in `test_threading_worker_execute` and `test_threading_worker_execute_with_dao` works for any type of worker. It anyway worked for `threading` and `subprocess`. But it fails with `redis` if `async` and `await` are not used.

Reply:
But I guess we can do away with `await` here. And we will still need to use `async` perhaps.